src/test/java/io/strimzi/kafka/bridge/http/AdminClientTest.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.bridge.http;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.strimzi.kafka.bridge.BridgeContentType;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.codec.BodyCodec;
+import io.vertx.junit5.VertxTestContext;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.is;
+
+public class AdminClientTest extends HttpBridgeTestBase  {
+    @Test
+    void listTopicsTest(VertxTestContext context) throws Exception {
+        String topic = "testListTopics";
+        kafkaCluster.createTopic(topic, 1, 1);
+
+        JsonArray records = new JsonArray();
+        JsonObject json = new JsonObject();
+        json.put("value", "hello");
+        records.add(json);
+
+        JsonObject root = new JsonObject();
+        root.put("records", records);
+
+        CompletableFuture<Boolean> producer = new CompletableFuture<>();
+        producerService()
+                .sendRecordsRequest(topic, root, BridgeContentType.KAFKA_JSON_JSON)
+                .sendJsonObject(root, ar1 -> {
+                    context.verify(() -> {
+                        assertThat(ar1.succeeded(), is(true));
+                        producer.complete(true);
+                    });
+                });
+        producer.get(TEST_TIMEOUT, TimeUnit.SECONDS);
```

Comment:
why you are producing message?

Reply:
kafkaCluster.createTopic at the beginning of the test adds a topic to Zk and it will take some time until the broker is notified/assigned to it. Sending a message and wait for its delivery as done here makes sure that the topic is known to the broker. Without this message sending and wait part, the test often fails with UnknownTopicOrPartitionException unless Thread.sleep is added to wait for a bit. Please let me know what we should do.

Reply:
Got it, let's stay with the producer.